corfudb-tools/src/main/java/org/corfudb/browser/CorfuStoreBrowser.java

```diff
@@ -83,27 +85,28 @@ public CorfuStoreBrowser(CorfuRuntime runtime) {
     public CorfuStoreBrowser(CorfuRuntime runtime, String diskPath) {
         this.runtime = runtime;
         this.diskPath = diskPath;
+        dynamicProtobufSerializer =
+            new DynamicProtobufSerializer(runtime);
+        Serializers.registerSerializer(dynamicProtobufSerializer);
     }
 
-    /**
-     * Validate that the namespace is not null
-     * @param namespace
-     */
-    private static void verifyNamespace(String namespace) {
-        if (namespace == null) {
-            throw new IllegalArgumentException("Please specify --namespace");
-        }
+    private boolean isEmptyOrNull(String str) {
+        return (str == null || str.isEmpty());
     }
 
     /**
      * Validate that both namespace and tablename are present
      * @param namespace - the namespace where the table belongs
      * @param tableName - table name without the namespace
      */
-    private static void verifyNamespaceAndTablename(String namespace, String tableName) {
-        verifyNamespace(namespace);
-        if (tableName == null) {
-            throw new IllegalArgumentException("Please specify --tablename");
+    private void verifyNamespaceAndTableName(String namespace,
+        String tableName, boolean tableNameValidationRequired) {
+        if (isEmptyOrNull(namespace)) {
+            throw new IllegalArgumentException("Namespace is empty or null");
+        }
+
+        if (tableNameValidationRequired && isEmptyOrNull(tableName)) {
+            throw new IllegalArgumentException("Tablename is empty or null");
```

Comment:
I know you didn't introduce this, but I think it should be removed. The parameter's should be validated once they are parsed from the command line and not "downstream". Move all these validations to `CorfuStoreBrowserMain` and just use `com.google.common.base.Preconditions.checkArgument` for validation.

Reply:
done